src/main/java/io/vertx/core/shareddata/ClusterSerializable.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2011-2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.shareddata;
+
+import io.vertx.core.buffer.Buffer;
+
+/**
+ * Objects implementing this interface will be written to and read from a {@link Buffer} when respectively
+ * stored and read from an {@link AsyncMap}.
+ * <p>
+ * Implementations must have a public no-argument constructor.
+ */
+public interface ClusterSerializable extends io.vertx.core.shareddata.impl.ClusterSerializable {
```

Comment:
I think it would be better the other way around : have the impl extends the non impl